zqd/api/api.go

```diff
@@ -2,11 +2,26 @@ package api
 
 import (
 	"encoding/json"
+	"errors"
+	"net/http"
+	"net/url"
 
 	"github.com/brimsec/zq/pkg/nano"
 	"github.com/brimsec/zq/zio/zjsonio"
+	"github.com/gorilla/mux"
 )
 
+// ExtractSpace returns the unescaped space from the path of a request.
+// XXX is this in the right package?
```

Comment:
In the existing package layout, I think this function should live in the space package.

Reply:
This can't go in package `space` because it's also needed in package `packet` and thus would cause a circular dependency. As `zqd` continues to grow I imagine we'll have some package dedicated to `http` helper functions (maybe extending mux.Router or something).

Reply:
But now that I'm thinking about this maybe the handler for `/space/{space}/packet` should live in `space` pkg, and deal with decoding the space path, and then call out the packet library with everything parsed.

Reply:
Alright I resolved this by moving the all endpoints into package `zqd` and made `ExtractSpace` an internal function of `zqd`